### ISSUE TITLE
Reverse tunnel

### ISSUE THREAD

**Issue by ghazel** [wangyu-/udp2raw-tunnel#320]
How can udp2raw be configured to run a reverse tunnel? For example, when the client can accept inbound connections, but the server cannot.

The example shows how to forward UDP packets sent to client port `3333` over a port `4096` tunnel to server port `7777`:

```
# Run at server side:
./udp2raw_amd64 -s -l0.0.0.0:4096 -r 127.0.0.1:7777    -k "passwd" --raw-mode faketcp -a

# Run at client side
./udp2raw_amd64 -c -l0.0.0.0:3333  -r44.55.66.77:4096  -k "passwd" --raw-mode faketcp -a
```

However, new packets sent to `127.0.0.1:7777` do not get sent to the client, because there is no `conv_id`.

**Comment by wangyu-** [comment]
udp2raw itself doesn't implement the function of reverse tunnel, but you can use udp2raw together with a vpn or a tunnel which supports reverse tunnel.

For example, with [udp2raw + openvpn](https://github.com/wangyu-/udp2raw-tunnel/wiki/udp2raw-openvpn-config-guide), once the vpn connection is established, upper-level connection can be establish in any direction.

(openvpn maybe too heavy weight for your purpose, I am just using it as an example here. you can replace openvpn with anything that supports reverse tunnel)

**Comment by ghazel** [comment]
The point of the reverse tunnel is to set up the vpn connection.

**Comment by wangyu-** [comment]
>the only way to establish a vpn connection would be to set up a reverse tunnel

no, you just run udp2raw server+vpn server on your client, and udp2raw client+vpn client on your server.  Once vpn connection is established, upper level connection can be established in any direction

**Comment by ghazel** [comment]
When trying to run udp2raw client on my server, I get `[FATAL]socket bind error`. Since the UDP port my VPN is listening on is already open.

**Comment by steom** [comment]
reverse tunnel is a most required feature for a cutting edge application

**Comment by wangyu-** [comment]
>reverse tunnel is a most required feature for a cutting edge application, please consider to implement in the next version.

I have already explained how to do reverse tunnel.

**Comment by steom** [comment]
Yes but with other programs.
That's not the point.

**Comment by a0s** [comment]
Hey! 1st of all, thanks for this nice software!!
I have the same question. I need to initiate udp2raw-connection from udp2raw-server-side (i have my server behind NAT and gray ip). What i want to do is expose port from inside NAT area to internet. For now i am using ssh port porwarding to external VPC server, but sometimes its unstable and says "port already used". Looks like udp2raw would help here..
